app/controllers/admin_controller.rb

```diff
@@ -1,5 +1,6 @@
 class AdminController < ApplicationController
   before_action :authorized?
+  load_and_authorize_resource
```

Comment:
Trả lời câu hỏi: Khai báo này dùng để làm gì?

Reply:
thay vì dùng the `authorize!` method để kiểm tra quyền cho từng action
thì mình dùng the `load_and_authorize_resource` method để tự động `load` cái resource và kiểm tra quyền cho tất cả action luôn
mục đích là tiết kiệm thời gian, đỡ phải viết đi viết lại mấy đoạn code

Reply:
Còn thiếu nhé tìm hiểu thêm

Reply:
bên trong thì nó sử dụng `before_action` 
`load_and_authorize_resource` là gộp của 2 câu lệnh rời: `load_resource` và `authorize_resource`
và nó có một quy tắc là cái controller sử dụng nó phải là RESTful style resource controller, nghĩa là các action của nó phải theo dạng: index, show, new, create, edit, update, destroy